[TSK-201] implement 3 different approaches to solve race condition (#2)
Hoocki/multithread: README.md

# Multithreading tasks
- [x] Heavy processing
- [x] Race condition
- [ ] Robot movement
- [ ] Payment deadlock
 ### Heavy processing
**Challenge:**

> We have method works 1 sec and this method is called N times. Change the code to finish all N tasks for about a second by using multithreading tools in parallel.\

Reviewer: ```suggestion
There is a method that does the job in 1 second and is called N times. Change the code to finish all N tasks for about a second by using multithreading tools in parallel.\
```

> This task is solved by implementing 2 approaches:

Reviewer: ```suggestion
Solution:
This task is solved by implementing 2 approaches:
```
add Solution for other tasks
